crates/driver/src/domain/competition/auction.rs

```diff
@@ -83,7 +83,7 @@ impl Auction {
     /// competition.
     pub async fn prioritize(mut self, eth: &Ethereum, balances: &eth::balances::Cache) -> Self {
         // Sort orders so that most likely to be fulfilled come first.
-        self.orders.sort_by_key(|order| {
+        self.orders.sort_by_cached_key(|order| {
```

Comment:
I threw this in on top.
Not really related to balance fetching but speeds up sorting from ~2.5s to ~0.2s and doesn't need a separate PR.